app/src/main/java/com/github/se/icebreakrr/ui/sections/shared/ProfileCard.kt

```diff
@@ -26,25 +27,77 @@ import androidx.compose.ui.unit.sp
 import coil.compose.AsyncImage
 import com.github.se.icebreakrr.R
 import com.github.se.icebreakrr.model.profile.Profile
+import com.google.accompanist.flowlayout.FlowRow
 
 // Define constants for layout dimensions and other configurations
 private val CARD_CORNER_RADIUS = 14.dp
-private val CARD_MAX_HEIGHT = 150.dp
-private val CARD_PADDING_HORIZONTAL = 10.dp
-private val CARD_PADDING_VERTICAL = 6.dp
+private val CARD_MAX_HEIGHT = 200.dp
+private val CARD_PADDING_HORIZONTAL = 20.dp
+private val CARD_PADDING_VERTICAL = 20.dp
+private val CARD_ELEVATION = 2.dp
 private val IMAGE_SIZE = 80.dp
 private val IMAGE_SPACING = 18.dp
-private val TEXT_SPACER_PADDING = 6.dp
+private val TEXT_SPACER_PADDING = 3.dp
 
 // Define constants for font sizes
-private val NAME_FONT_SIZE = 24.sp
+private val NAME_FONT_SIZE = 18.sp
 private val CATCHPHRASE_FONT_SIZE = 16.sp
-private val TAGS_FONT_SIZE = 16.sp
-
-// Define colors used
+private val TAGS_FONT_SIZE = 10.sp
 
 // Define math constants
-private val TAKE_TAGS = 5
+private val TAKE_TAGS = 3
+
+@Composable
+fun TagDisplay(tags: List<String>, isSettings: Boolean) {
+  if (!isSettings) {
+    // Determine the number of tags to display
+    val tagsToDisplay =
+        when {
+          tags.size > TAKE_TAGS -> TAKE_TAGS - 1 // Display only 2 if there are more than 3 tags
+          else -> tags.size // Otherwise, display all available tags
+        }
+
+    // Display the tags with a background and spacing
+    FlowRow(modifier = Modifier.fillMaxWidth().padding(TEXT_SPACER_PADDING)) {
+      // Create a Text for each tag, limiting to the calculated number of tags
+      tags.take(tagsToDisplay).forEachIndexed { index, tag ->
+        Text(
+            text = "#$tag",
+            fontSize = TAGS_FONT_SIZE,
+            fontWeight = FontWeight.SemiBold,
+            color = MaterialTheme.colorScheme.primaryContainer,
+            modifier =
+                Modifier.padding(2.dp)
+                    .clip(RoundedCornerShape(25.dp)) // Rounded corners for each tag
+                    .background(
+                        MaterialTheme.colorScheme.secondaryContainer) // Tag background color
+                    .padding(horizontal = 10.dp, vertical = 4.dp) // Padding inside the tag
+            )
+      }
+
+      // If there are more than 3 tags, add "..." at the end
+      if (tags.size > TAKE_TAGS) {
+        Text(
+            text = "...",
+            fontSize = TAGS_FONT_SIZE,
+            fontWeight = FontWeight.SemiBold,
+            color = MaterialTheme.colorScheme.primaryContainer,
+            modifier =
+                Modifier.padding(2.dp)
+                    .clip(RoundedCornerShape(25.dp)) // Rounded corners for the ellipsis
```

Comment:
Nitpick: magic number

Reply:
This is too much ! My system need a reboot to focus on those MaGiCC Nu..Nu..mbers 🤖⚡️